torch/distributed/_tensor/api.py

```diff
@@ -253,6 +253,10 @@ def __tensor_unflatten__(inner_tensors, flatten_spec):
     # pyre-fixme[3]: Return type must be annotated.
     # pyre-fixme[2]: Parameter must be annotated.
     def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+        # Decomposes CompositeImplicitAutograd ops
+        r = func.decompose(*args, **kwargs)
```

Comment:
We can't by default do this, this would bring quite a bit CPU overhead I believe, we should hide this under a flag check. @bdhirsh @Chillee trying to understand this code https://github.com/pytorch/pytorch/blob/main/torch/_ops.py#L539 and I want to see if there's a simple way to check if op need to be decomposed or not, and call this under this flag. Otherwise we have to either implement a linear op rule, or we do special check for `aten.linear.default`

Reply:
Add a direct op check and synced with @wanchaol offline, we don't want to use an op allowlist here.